src/NHibernate/Linq/Visitors/NullableExpressionDetector.cs

```diff
@@ -128,6 +128,13 @@ private bool IsNullable(MemberExpression memberExpression, BinaryExpression equa
 		{
 			if (_functionRegistry.TryGetGenerator(memberExpression.Member, out _))
 			{
+				// The expression can be null when the member is static (e.g. DateTime.Now).
+				// In such cases we suppose that the value cannot be null.
+				if (memberExpression.Expression == null)
+				{
+					return false;
```

Comment:
What if a Linq query references a static member of the class in which it is coded?
```
public SomeClass
{
  private int? _someValue;

  public void SomeMethod()
  {
    var results = session.Query<Bidule>().Where(b => b.Truc != _someValue).ToList();
  }
}
```

We should probably check what is the member type nullability, or assume it to be nullable for staying on the safe side.

It also seems to me this fix may deserve a PR on its own, especially if this one cannot be merged soon.

Reply:
> What if a Linq query references a static member of the class in which it is coded?

Then it will be pre-evaluated and treated as a parameter, as far as I know the only static properties that are not pre-evaluated are introduced with this PR. Also `NoPreEvaluationAttribute` works only for methods so I think we should be good by saying that they are not nullable.

Reply:
The only way to create a nullable static property that is not pre-evaluated is by creating a custom `IHqlGeneratorForProperty` and implement `IAllowPreEvaluationHqlGenerator` that is introduced by this PR. If we want to have this configurable we could extend `IHqlGeneratorForProperty` similar as we extended `IHqlGeneratorForMethod` by adding `AllowsNullableReturnType` method.

Reply:
I had forget this, you are right. This PR is a year and a half old already.